AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -189,11 +196,9 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
     }
     
     private func enqueueForSaving(measurements: SDCardMeasurementsRow, buffer streamsWithMeasurements: inout [SDStream: [SDSyncMeasurement]], deviceID: String) {
-        var location: CLLocationCoordinate2D?
-        if let lat = measurements.lat, let long = measurements.long {
-            location = CLLocationCoordinate2D(latitude: lat, longitude: long)
+        if location == nil {
+            location = addMeasurementLocation(deviceID: deviceID, measurements: measurements)
         }
```

Comment:
That's mine, right? It was changed to:  
`location = location ?? setProperLocation(deviceID: deviceID, measurements: measurements)
`

Reply:
Yes, mine
<img width="454" alt="Screenshot 2024-04-17 at 11 45 13" src="https://github.com/HabitatMap/AirCastingiOS/assets/46443841/0b1d3e98-6614-4ddd-90ad-2583dc742adf">